src/main/java/org/codeforamerica/shiba/application/parsers/ApplicationDataParser.java

```diff
@@ -11,11 +12,11 @@ public ApplicationDataParser(ParsingConfiguration parsingConfiguration) {
 
     public abstract T parse(ApplicationData applicationData);
 
-    protected static double getDouble(ApplicationData applicationData, PageInputCoordinates pageInputCoordinates) {
+    protected static Money getMoney(ApplicationData applicationData, PageInputCoordinates pageInputCoordinates) {
         try {
-            return Double.parseDouble(applicationData.getValue(pageInputCoordinates).replace(",",""));
+            return new Money(Double.parseDouble(applicationData.getValue(pageInputCoordinates).replace(",","")));
```

Comment:
I wonder if we should avoid parsing doubles at all, and instead have the Money constructor accept a string

[That article you shared in tracker](https://cdn.app.compendium.com/uploads/user/e7c690e8-6ff9-102a-ac6d-e4aebca50425/34d3828b-c12e-4c7e-8942-b6b7deb02e12/File/8712e504216fcf1c3b5f37402c013172/four_common_pitfalls_of_the_bigdecimal_class_and_how_to_avoid_them.pdf) about BigDecimal suggests that using the `double` constructor of `BigDecimal` can cause precision issues since now we went into floating point land for a while

Reply:
If we do need to parse as a double for whatever reason, we should use `BigDecimal.valueOf(double)`